syntax = "proto3";

message Robots_Status{
  repeated Robot_Status robots_status = 1;
}

message Robot_Status {
  uint32 robot_id  = 1;
  bool infrared   = 2;
  bool flat_kick  = 3;
  bool chip_kick  = 4;
}
message Robots_Command {
  repeated Robot_Command command = 1;

  // delay * 0.1ms
  uint32 delay = 2;
}

message Robot_Command {
  // The unique ID of the robot, as identified by SSL-Vision.
  uint32 robot_id = 1;

  // Desired forward drive velocity in cm/s .
  float velocity_x = 2;

  // Desired sideways left drive velocity in cm/s .
  float velocity_y = 3;

  // Desired counterclockwise angular velocity in 1/40 radians / second.
  float velocity_r = 4;

  // 0 for flat kick , 1 for chip kick .
  bool kick = 5;
  
  // Desired flat kick speed, in cm/s.
  // If this field is missing, then no flat kick command should be
  // sent.
  // Desired chip kick distance, in cm.
  // If this field is missing, then no flat kick command should be
  // sent. If this field is present along with the flat_kick field, the
  // flat_kick command takes precedence.
  float power = 6;

  float dribbler_spin = 7;

  // new protocol
  float current_angle = 8;
  float target_angle = 9;
}